src/grid/mod.rs

```diff
@@ -240,6 +243,13 @@ impl<T: Copy + Clone> Grid<T> {
         self.raw.set_visible_lines(new_line_count);
         self.lines = new_line_count;
 
+        // Fill up the history with empty lines
+        if self.raw.len() < self.raw.capacity() {
+            for _ in self.raw.len()..self.raw.capacity() {
+                self.raw.push(Row::new(self.cols, &template));
+            }
+        }
```

Comment:
Correct me if I'm wrong but this is a temparary solution until we have a buffer that grows and shrinks with the needed size. See #1236